src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsnonassociationsapi/resource/NonAssociationsResource.kt

```diff
@@ -216,6 +227,83 @@ class NonAssociationsResource(
     )
   }
 
+  @GetMapping("/non-associations/between")
+  @PreAuthorize("hasRole('ROLE_NON_ASSOCIATIONS')")
+  @ResponseStatus(HttpStatus.OK)
+  @Operation(
+    summary = "Get a non-associations between two prisoners by prisoner number.",
+    description = "Requires ROLE_NON_ASSOCIATIONS role.",
+    responses = [
+      ApiResponse(
+        responseCode = "200",
+        description = "Returns the non-associations",
+      ),
+      ApiResponse(
+        responseCode = "400",
+        description = "When two distinct prisoner numbers aren't provided or neither open nor closed non-associations are included",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))],
+      ),
+      ApiResponse(
+        responseCode = "401",
+        description = "Unauthorized to access this endpoint",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))],
+      ),
+      ApiResponse(
+        responseCode = "403",
+        description = "Missing required role. Requires the NON_ASSOCIATIONS role",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))],
+      ),
+    ],
+  )
+  fun getNonAssociationsBetweenPrisoners(
+    @Schema(description = "A prisoner number", required = true, example = "A1234BC")
+    @RequestParam(required = false)
+    firstPrisonerNumber: String?,
+
+    @Schema(description = "Another prisoner number", required = true, example = "A1234BC")
+    @RequestParam(required = false)
+    secondPrisonerNumber: String?,
+
+    @Schema(
+      description = "Whether to include open non-associations or not",
+      required = false,
+      defaultValue = "true",
+      example = "false",
+    )
+    @RequestParam(required = false, defaultValue = "true")
```

Comment:
Shouldn't these be required = true

Reply:
`firstPrisonerNumber` and `secondPrisonerNumber` are technically required (this is reflected in swagger docs) but if you make them required in the `RequestParam` annotation, then you get an unhelpful 500 internal server error if they're absent. hence the manual check in the method body that returns a 400 instead since the _user_ request wasn't valid.

`includeClosed` and `includeOpen` are genuinely not required since there's a sensible default we can use